src/smb_share.c

```diff
@@ -179,9 +191,21 @@ static ssize_t  smb_share_parse_enum(smb_message *msg, char ***list)
     {
         uint32_t name_len, com_len;
 
+        if (data + sizeof(uint32_t) > eod)
```

Comment:
This also seems easier to understand if written as something like `msg->payload_size < 72 + share_count * 1` as the error condition, plus it could be removed from the loop
Although I'm not entirely sure how to understand the various offset adjustments below, though since we're not re-reading from the network, I suppose it should work

Reply:
The `name_len` is given by the server, so we do not know the size of all share name without parse the message. So I dont think we can move this check out of the loop. But may be we can use a separated loop to check the message size, then use another loop to parse the share names.